bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs

```diff
@@ -23,14 +23,52 @@ namespace MongoDB.Libmongocrypt
     /// </summary>
     public class CryptClientFactory
     {
+        // MUST be static fields since otherwise these callbacks can be collected via the garbage collector
+        // regardless they're used by mongocrypt level or no
+        private static Library.Delegates.CryptoCallback __crypto256DecryptCallback = new Library.Delegates.CryptoCallback(CipherCallbacks.Decrypt);
+        private static Library.Delegates.CryptoCallback __crypto256EncryptCallback = new Library.Delegates.CryptoCallback(CipherCallbacks.Encrypt);
+        private static Library.Delegates.HashCallback __cryptoHashCallback = new Library.Delegates.HashCallback(HashCallback.Hash);
+        private static Library.Delegates.CryptoHmacCallback __cryptoHmacSha256Callback = new Library.Delegates.CryptoHmacCallback(HmacShaCallbacks.HmacSha256);
+        private static Library.Delegates.CryptoHmacCallback __cryptoHmacSha512Callback = new Library.Delegates.CryptoHmacCallback(HmacShaCallbacks.HmacSha512);
+        private static Library.Delegates.RandomCallback __randomCallback = new Library.Delegates.RandomCallback(SecureRandomCallback.GenerateRandom);
+        private static Library.Delegates.CryptoHmacCallback __signRsaesPkcs1HmacCallback = new Library.Delegates.CryptoHmacCallback(SigningRSAESPKCSCallback.RsaSign);
+
         /// <summary>Creates a CryptClient with the specified options.</summary>
         /// <param name="options">The options.</param>
         /// <returns>A CryptClient</returns>
         public static CryptClient Create(CryptOptions options)
         {
-            MongoCryptSafeHandle handle = Library.mongocrypt_new();
+            var handle = Library.mongocrypt_new();
+
+            var status = new Status();
 
-            Status status = new Status();
+            // The below code can be avoided on Windows. So, we don't call it on this system 
+            // to avoid restrictions on target frameworks that present in some of below
+            if (OperatingSystemHelper.CurrentOperatingSystem != OperatingSystemPlatform.Windows)
+            {
+#if !NETSTANDARD1_5
```

Comment:
these `ifdef` conditions are a bit tricky. The reason why we added them is that we want to provide the ability for users to manually provide the `mongocrypt` library which will include a linked OpenSSL library which can be used instead of the added hooks for target frameworks which we don't support. So, we need to avoid exceptions in this case.
This behavior should be tested more deeply on Linux. So, I will test it in the next ticket